airbyte-webapp/src/views/Connector/ConnectorCard/ConnectorCard.tsx

```diff
@@ -1,128 +1,121 @@
-  const { trackTestConnectorFailure, trackTestConnectorSuccess, trackTestConnectorStarted } =
-    useAnalyticsTrackFunctions(props.formType);
-
-  useEffect(() => {
-    // Whenever the selected connector changed, reset the check connection call and other errors
-    reset();
-    setErrorStatusRequest(null);
-  }, [props.selectedConnectorDefinitionSpecification, reset]);
-
-  const onHandleSubmit = async (values: ServiceFormValues) => {
-    setErrorStatusRequest(null);
-    setIsFormSubmitting(true);
-
-    const connector = props.availableServices.find((item) => Connector.id(item) === values.serviceType);
-
-    const testConnectorWithTracking = async () => {
-      trackTestConnectorStarted(connector);
-      try {
-        await testConnector(values);
-        trackTestConnectorSuccess(connector);
-      } catch (e) {
-        trackTestConnectorFailure(connector);
-        throw e;
-      }
-    };
-
-    try {
-      await testConnectorWithTracking();
-      onSubmit(values);
-      setSaved(true);
-    } catch (e) {
-      setErrorStatusRequest(e);
-      setIsFormSubmitting(false);
-    }
-  };
-
-  const job = jobInfo || LogsRequestError.extractJobInfo(errorStatusRequest);
-
-  const { selectedConnectorDefinitionSpecification, onServiceSelect, availableServices, isEditMode } = props;
-  const selectedConnectorDefinitionSpecificationId =
-    selectedConnectorDefinitionSpecification && ConnectorSpecification.id(selectedConnectorDefinitionSpecification);
+import { ConnectorCardProps } from "./interfaces";
+import { useConnectorCardService } from "./useConnectorCardService";
+
+export const ConnectorCard: React.FC<ConnectorCardProps> = (props) => {
+  const {
+    title,
+    full,
+    isLoading,
+    additionalSelectorComponent,
+    formType,
+    selectedConnectorDefinitionSpecification,
+    isEditMode,
+    availableServices,
+    onServiceSelect,
+    onDelete,
+  } = props;
+  const {
+    advancedMode,
+    error,
+    formFields,
+    getValues,
+    initialValues,
+    isFormSubmitting,
+    isTestConnectionInProgress,
+    job,
+    jsonSchema,
+    onFormSubmit,
+    onHandleSubmit,
+    onStopTesting,
+    resetUiWidgetsInfo,
+    saved,
+    selectedConnectorDefinitionSpecificationId,
+    setUiWidgetsInfo,
+    testConnector,
+    uiWidgetsInfo,
+    uniqueFormId,
+    validationSchema,
+  } = useConnectorCardService(props);
 
   return (
-    <Card title={title} fullWidth={full}>
-      <div className={styles.cardForm}>
-        <div className={styles.connectorSelectControl}>
-          <ConnectorServiceTypeControl
-            formType={props.formType}
-            onChangeServiceType={onServiceSelect}
-            availableServices={availableServices}
-            isEditMode={isEditMode}
-            selectedServiceId={selectedConnectorDefinitionSpecificationId}
-            disabled={isFormSubmitting}
-          />
-        </div>
-        {additionalSelectorComponent}
+    <Formik
```

Comment:
Initially I thought it felt slightly weird to have the `Formik` be declared outside of `ServiceForm`, but given that the goal of this is to move the "controls" outside of the card itself, then this structure probably makes sense 👍